sw/device/tests/aes_force_prng_reseed_test.c

```diff
@@ -0,0 +1,271 @@
+void generate_new_key(dif_aes_key_share_t *key, int index) {
+  uint8_t new_key_share0[sizeof(kAesModesKey128)];
+  // Generate new key share0 by XOR-ing base key with kKeyShare1*index
+  for (size_t i = 0; i < sizeof(kAesModesKey128); ++i) {
+    new_key_share0[i] = kAesModesKey128[i] ^ kKeyShare1[i] * (uint8_t)(index);
+  }
+  // Update the key shares
+  memcpy(key->share0, new_key_share0, sizeof(key->share0));
+  memcpy(key->share1, kKeyShare1, sizeof(key->share1));
+}
+
+status_t execute_test(void) {
+  bool aes_idle = false;
+  bool input_ready = false;
+  bool output_valid = false;
+
+  LOG_INFO(
+      "Testing aes_prng_reseed_test with number of blocks: %d (Block 0 to %d), "
+      "generating new key at block: %d and disabling entropy at start enable"
+      ": %d and disabling entropy after block: %d",
+      kAesNumBlocks, kAesNumBlocks - 1, generate_new_key_at_block - 1,
+      disable_entropy_at_start_en, disable_entropy_after_block - 1);
+  // Initialize AES
+  CHECK_DIF_OK(
+      dif_aes_init(mmio_region_from_addr(TOP_EARLGREY_AES_BASE_ADDR), &aes));
+  CHECK_DIF_OK(dif_aes_reset(&aes));
+  // Initialize EDN0, EDN1, CSRNG and Entropy Source
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN0_BASE_ADDR), &edn0));
+  CHECK_DIF_OK(
+      dif_edn_init(mmio_region_from_addr(TOP_EARLGREY_EDN1_BASE_ADDR), &edn1));
+  CHECK_DIF_OK(dif_csrng_init(
+      mmio_region_from_addr(TOP_EARLGREY_CSRNG_BASE_ADDR), &csrng));
+  CHECK_DIF_OK(dif_entropy_src_init(
+      mmio_region_from_addr(TOP_EARLGREY_ENTROPY_SRC_BASE_ADDR), &entropy_src));
+
+  // Generate key with index 0
+  generate_new_key(&key, 0);
+
+  // Prepare transaction
+  transaction.operation = kDifAesOperationEncrypt;
+  transaction.mode = kDifAesModeEcb;
+  transaction.key_len = kDifAesKey128;
+  transaction.key_provider = kDifAesKeySoftwareProvided;
+  transaction.mask_reseeding = kDifAesReseedPerBlock;
+  transaction.manual_operation = kDifAesManualOperationManual;
+  transaction.reseed_on_key_change = true;
+  transaction.force_masks = true;
+  transaction.ctrl_aux_lock = false;
+
+  if (disable_entropy_at_start_en == 1) {
+    LOG_INFO(
+        "Disabling entropy complex to induce AES stall before starting AES");
+    CHECK_STATUS_OK(disable_entropy_complex());
+    busy_spin_micros(500);
+  }
+
+  // Start the AES operation
+  // Wait for AES to be idle before starting encryption
+  aes_idle = false;
+  do {
+    CHECK_DIF_OK(dif_aes_get_status(&aes, kDifAesStatusIdle, &aes_idle));
+  } while (!aes_idle);
+
+  CHECK_DIF_OK(dif_aes_start(&aes, &transaction, &key, NULL));
+
+  dif_aes_data_t plain_text[kAesNumBlocks];
+  dif_aes_data_t cipher_text[kAesNumBlocks];
+
+  // Initialize plaintext data dynamically
+  // Create plaintext with incremental data
+  for (int i = 0; i < kAesNumBlocks; ++i) {
+    for (int j = 0; j < kDifAesBlockNumBytes; ++j) {
+      plain_text[i].data[j] = (uint8_t)((i * kDifAesBlockNumBytes + j) & 0xFF);
```

Comment:
Maybe use entropy like here:
https://github.com/lowRISC/opentitan/blob/831113c05ab4620c551806e26ba6963e50eec910/sw/device/tests/pmod/i2c_host_fram_test.c#L145-L149

Reply:
Updated code snippet::

#include "sw/device/lib/dif/dif_rv_core_ibex.h"
#include "sw/device/lib/testing/rv_core_ibex_testutils.h"
  
  // Initialize plaintext data dynamically
  // Create plaintext with random data
  dif_rv_core_ibex_t rv_core_ibex;
  CHECK_DIF_OK(dif_rv_core_ibex_init(
      mmio_region_from_addr(TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR),
      &rv_core_ibex));
  for (int i = 0; i < kAesNumBlocks; ++i) {
    for (int j = 0; j < kDifAesBlockNumBytes; ++j) {
      uint32_t rand;
      TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, 2000, &rand));
      plain_text[i].data[j] = rand;
    }
  }

Reply:
The snippet is working code.  Please let me know your views on it.

Reply:
Code updated and checked in

Reply:
I don't see the change.

Reply:
Done